### DISCUSSION TITLE
Clarification: Understanding Shape of locations

### DISCUSSION THREAD

**Discussion by cece3014** [talmolab/sleap#889] [Help!]
Hi All,

Thanks for all your help so far! I just wanted to ask about the parameters of the shape of the locations dataset.
My first parameter is coming out as 238 frames, but in the GUI I have 1986 frames, with 449 predicted frames. I am just curious as to what the 238 could be referring to? I intend on extracting the x,y coordinates of my nodes and want to ensure I do so for all my labeled and predicted frames. Thank you!

**Comment by roomrys** [comment]
Hi @cece3014,

The first shape of the [`locations` matrix](https://sleap.ai/notebooks/Analysis_examples.html#loading-the-data:~:text=Here%E2%80%99s%20what%20each%20dimension%20of%20the%20matrix%20means%3A) is the frame count. Does your project happen to have multiple videos? If so, one analysis file is output per video which could be why there are less frames than expected.

Let us know if that is not the case and we will dig into it further.

Thanks,
Liezl

**Reply by cece3014** [reply]
Hi! Thank you for the quick response! yes, my project has multiple videos, so that's great to know! However. none of them have 238 frames, I am curious as to what frame count this number evaluates.

**Reply by roomrys** [reply]
Hi @cece3014,

The default for exporting to analysis files is to export locations for frames starting at the beginning of the video and ending at the last frame where a prediction/label was found. Do you have a last label/prediction on frame 238?

https://github.com/talmolab/sleap/blob/740f9fa7b1d5a4640cf8d96b6e4185273219346a/sleap/info/write_tracking_h5.py#L102-L106

Likewise, if this isn't the case, feel free to keep reaching out.

Thanks,
Liezl

**Reply by cece3014** [reply]
Hi Liezel,
I have found the video where there is a final prediction on a frame 238! Thank you.
However, when I attempt to highlight a different video, and export the HDF5 file, the shape of the location matrix continue to claim 238(even if the mac is 118). I am not sure how to resolve this, do you have any recommendations?

**Reply by roomrys** [reply]
Hi @cece3014,

Glad to help!

You need to have the video loaded as the current video in the GUI, i.e. you need to double click on the video in the Videos dialog - not just have it highlighted.

Let us know if that works.

Thanks,
Liezl

**Reply by cece3014** [reply]
Hi Liezl,
Thank you for your response! I have double clicked the first video in the GUI, exported the HDF5 file, but the frame number still doesn't match the number of frames in the video (video has 118, but now the matrix claims 254?)
254 is the max number of frames i have in my 13th video (13/14 total), could the matrix just be stating the max number of frame in the project rather than of a video?

Not sure how to fix this, would I need to only have one video per project? As in, make copies of my main project, and create one project per video?

Thank you!

**Reply by roomrys** [reply]
Hi @cece3014,

This is definitely strange behavior. I double checked that things work as expected on my local install and things seem to run as expected over here. We added a feature to support exporting analysis files for multi-video projects in [Exporting analysis HDF5 files for multi-video projects](https://github.com/talmolab/sleap/issues/729) which was integrated in SLEAP v1.2.4. Are you running SLEAP v1.2.4 or above?

Just to make sure:
For the video stats in the seek bar screenshot below,

![image](https://user-images.githubusercontent.com/38435167/183133302-75987578-dcb3-4b1a-a0e6-501cfd872b2c.png)
**Fig 1:** The video that is loaded in the GUI will pop-up with some stats, namely, we can see that the locations matrix should have 399 frames.

if I check "Export Analysis HDF5 > Current Video...", I would expect an analysis file with a locations matrix of 399 frames.

![image](https://user-images.githubusercontent.com/38435167/183133688-825564b0-13f2-4839-9403-f0f7d5e942ab.png)
**Fig 2:** SLEAP v1.2.4 and above gives an option to export an analysis file for the current video.

Sorry, I should have checked that you were on the correct version of SLEAP from the get-go. If that still isn't the case, is it possible to share your dataset with me (or a subset) at [lmaree@salk.edu](mailto:lmaree@salk.edu).

Thanks and sorry for all the trouble you're experiencing!
Liezl
